lib/logstash/outputs/loggly.rb

```diff
@@ -88,31 +110,59 @@ class LogStash::Outputs::Loggly < LogStash::Outputs::Base
 
   public
   def register
-    # nothing to do
+    buffer_initialize(
+      :max_items => @flush_size,
+      :max_interval => @idle_flush_time,
+      :logger => @logger
+    )
   end
 
   public
   def receive(event)
     key = event.sprintf(@key)
     tag = event.sprintf(@tag)
 
+    if expected_field = key[/%{(.*)}/, 1]
+      @logger.warn "Skipping sending message to Loggly. No key provided (key='#{key}'). Make sure to set field '#{expected_field}'."
+      @logger.info "Dropped message: #{event.to_json}"
```

Comment:
let's make this a debug level message, with big enough events this could flood the logs.
Also, we typically prefer structured logging instead of interpolation, so we can do this instead:

`@logger.info("Dropped message", :event => event.to_json)`

Reply:
Oh yes, both suggestions make total sense.